Minor language corrections/improvements (#3392)
qgis/QGIS-Documentation: source/docs/user_manual/introduction/general_tools.rst

* derived measurements in the :ref:`identify` tool
* the vector analysis tool: :menuselection:`Vector --> Geometry Tools -->
  Export/Add Geometry Columns`

Measuring works within projected coordinate systems (e.g., UTM) and unprojected
data. The first three measuring tools behave equally to global project settings:

* Unlike most other GIS, the default measurement metric is
  ellipsoidal, using the ellipsoid defined in
  :menuselection:`Project --> Properties... --> General`. This is true both
  when geographic and projected coordinate systems are defined for the project.
* If you want to calculate the projected/planimetric area or distance using cartesian
  maths, the measurement ellipsoid has to be set to "None/Planimetric"
  (:menuselection:`Project --> Properties... --> CRS`). However,
  with a geographic (ie unprojected) CRS defined for the data and project, area and
  distance measurement will be ellipsoidal.

However, neither the identify tool nor the field calculator will transform your
data to the project CRS before measuring. If you want to achieve this, you have
to use the vector analysis tool: :menuselection:`Vector --> Geometry Tools -->

> Add Geometry Attributes...`. Here, measurement is by default planimetric, unless

Reviewer: ```suggestion
Add Geometry Attributes...`. Here, measurement is planimetric, unless
```

Reviewer: Too much with both "by default" and "unless"?

Author: Agreed.

> Export/Add Geometry Columns`. Here, measurement is by default planimetric, unless

Reviewer 2: Renamed.
```suggestion
Add Geometry Attributes...`. Here, measurement is by default planimetric, unless
```

For string comparisons, it is also possible to use the |checkbox|
:guilabel:`Case sensitive` option.

After setting all search options, click :guilabel:`Select features`
to select the matching features. The drop-down options are:

* :guilabel:`Select features`
* :guilabel:`Add to current selection`
* :guilabel:`Filter current selection`
* :guilabel:`Remove from current current selection`

You can also clear all search options using the :guilabel:`Reset form` button.

Once the conditions are set, you can also either:

* :guilabel:`Zoom to features` on the map canvas without the need of a preselection
* :guilabel:`Flash features`, highlighting the matching features. This is a
  handy way to identify a feature without selection or using the Identify tool.
  Note that the flash does not alter the map canvas extent and would be visible only

>   if the feature is in within the bounds of the current map canvas.

Reviewer: ```suggestion
  if the feature is within the bounds of the current map canvas.
```

Author: Thank you.

  is added, namely ``View feature form`` for editing. You can define more actions
  in the layer's properties dialog (see :ref:`actions_menu`).
* **Derived**: This information is calculated or derived from other information.
  It includes:

  * general information about the feature's geometry:

    * depending on the geometry type, the cartesian measurements of length,
      perimeter or area in the layer's CRS units
    * depending on the geometry type and if an ellipsoid is set in the project
      properties dialog for :guilabel:`Measurements`, the ellipsoidal values of
      length, perimeter or area using the specified units
    * the count of geometry parts in the feature and the number of the part
      clicked
    * the count of vertices in the feature
  * coordinate information, using the project properties :guilabel:`Coordinates
    display` settings:

    * ``X`` and ``Y`` coordinate values of the point clicked
    * the number of the closest vertex to the point clicked
    * ``X`` and ``Y`` coordinate values of the
      closest vertex (and ``Z`/`M`` if applicable)

>     * if click on a curved line,

Reviewer: ```suggestion
    * if a curved line is clicked,
```

Reviewer: I don't know if this is better?

Author: It's worse - I got this wrong. I'll rewrite.

Author: ```suggestion
    * if the geometry is a curve,
```

Reviewer: Fine.  But how does this relate to "the radius of that section"?  Where does "section" come from?

Reviewer 2: curved segment? Segment is often used in the dialog (eg for snapping) and its refers to the line between two consecutive vertex

Reviewer: @DelazJ, segment is a good term, but a circular string is defined by its two end "vertices", but in addition there is a third vertex that controls the radius of the circular string (try the vertex tool to see the effect).

Reviewer: I thought that QGIS did not support other curved geometries than circular string.  If it does, we need to be more generic.  But we are talking about a circular string as **a part** of a geometry. Many geometry types can contain circular string parts.

Author: It definitely supports them *to some extent*. When they were first introduced, though, they could not be ?displayed/?edited. I don't know what the current position is. Perhaps I should ask the dev list for a steer on this?

Reviewer: I guess that `circular string` comes from SQL MM/Spatial. In that specification, the following geometry types for lines (1D geometries) are included: Curve, LineString, CircularString, Circle, GeodesicString, EllipticalCurve, NURBSCurve, Clothoid, SpiralCurve, CompoundCurve + MultiCurve and MultiLineString.
Curve is abstract (superclass of LineString, CircularString, Circle (that is a subtype of CircularString), GeodesicString, EllipticalCurve, NURBSCurve, Clothoid, SpiralCurve, CompoundCurve (that is a sequence of curves joined at the end points)).
The question to the developers would be if QGIS supports other basic line geometries than LineString and CircularString (including Circle).

Anyway, since `curve` also includes LineString, it should not be used here.

Reviewer: Just saw your message to the devs list - excellent!

Author: No response from the dev list. Perhaps we should just merge and move on at this stage?

Author: It's meaningful for any curved segment: https://en.wikipedia.org/wiki/Radius_of_curvature. To get rid of thee passive from your suggestion, and the assertion that the geometry type must be a cuvre, how about "if you click on a curve/curved line/curved segment"?

Reviewer: Your suggestion is fine, as long as "curve" is removed, since curve (in the SQL MM/Spatial model) is a generalisation of all kinds of lines, including polylines (LineString), for which QGIS will not give you a radius.

Author: I meant them as alternatives for the text, not all to be included. So which do we prefer, "curved line" or "curved segment"?

Reviewer: I see. I prefer "curved segment" (or "curved line segment").

Author: Me too. Thanks.

Author: ```suggestion
    * if you click on a curved segment,
```

Reviewer: Just do it :-)

>       the radius of that section is also displayed.

Reviewer: ```suggestion
      and the click is in the middle of that element, its radius is also displayed.
```

Reviewer: I just tested this in QGIS 3.4.4.  You need to click closer to the second point that defines the circular string than to the ends in order for the radius to be displayed.

Reviewer 2: "and the click is in the middle of that element". This condition does not apply. Radius seems to be always there as column. And only filled for vertices that are on a curved segment.

Reviewer 2: I really should go and sleep. I don't know why I was trying with the vertex tool instead of the identify one. /me going away....

Reviewer: That is what I indicated in my previous comment.
In my "Identify Results" window, I only see the "Closest vertex radius" line when my click is closest to the middle defining point of the circular string.  I don't see a radius column, so I guess we are looking at different things.

>       the radius of that section in the panel result is also displayed.

Reviewer: ```suggestion
      the radius of that section is also displayed.
```

Author: Good.

.. _identify_mode:

At the bottom of the window are the :guilabel:`Mode` and :guilabel:`View`
comboboxes.
:guilabel:`Mode` defines from which layers features should be identified:

* **Current layer**: only features from the selected layer are identified. The
  layer need not be visible in the canvas.
* **Top down, stop at first**: only features from the upper visible layer.
* **Top down**: all features from the visible layers. The results are shown in
  the panel.

> * and **Layer selection**: opens a context menu where the user selects the layer to

Reviewer: ```suggestion
* **Layer selection**: opens a context menu where the user selects the layer to
```

Author: Thank you.

>   identify features from, similar to like a right-click. Only the chosen features

Reviewer: ```suggestion
  identify features from, similar to a right-click. Only the chosen features
```

Author: Thanks - whoops.
